From 84d3555ff56b5fc5535fd2bb9e0fe8c0d10f9646 Mon Sep 17 00:00:00 2001
From: Jean Trivelly <jean.trivelly@intel.com>
Date: Mon, 25 Feb 2013 18:03:41 +0100
Subject: [PORT FROM MAIN] wifi: Add Freq info in STATUS command in CLI.

BZ: 89448

Add Freq info in STATUS command in CLI.

Feature   : Wlan
Category  : Improvement
Purpose   : AOSP+
Origin    : Upstream
Upstream-candidate: yes

Change-Id: Ic5440df77c727bdbedabbf2d7d0ddfc191257cbe
Orig-Change-Id: If1cedd39ec1dd2f37661c733934dc97742f48ec1
Signed-off-by: Jean Trivelly <jean.trivelly@intel.com>
---
 wpa_supplicant/ctrl_iface.c |   23 +++++++++++++++++++++++
 1 files changed, 23 insertions(+), 0 deletions(-)

diff --git a/wpa_supplicant/ctrl_iface.c b/wpa_supplicant/ctrl_iface.c
index 35769f8..e5393ed 100644
--- a/wpa_supplicant/ctrl_iface.c
+++ b/wpa_supplicant/ctrl_iface.c
@@ -1408,6 +1408,7 @@ static int wpa_supplicant_ctrl_iface_status(struct wpa_supplicant *wpa_s,
 	end = buf + buflen;
 	if (wpa_s->wpa_state >= WPA_ASSOCIATED) {
 		struct wpa_ssid *ssid = wpa_s->current_ssid;
+		struct wpa_bss *bss = wpa_s->current_bss;
 		ret = os_snprintf(pos, end - pos, "bssid=" MACSTR "\n",
 				  MAC2STR(wpa_s->bssid));
 		if (ret < 0 || ret >= end - pos)
@@ -1481,6 +1482,28 @@ static int wpa_supplicant_ctrl_iface_status(struct wpa_supplicant *wpa_s,
 			if (ret < 0 || ret >= end - pos)
 				return pos - buf;
 			pos += ret;
+
+			/* Add Freq */
+			switch (ssid->mode) {
+			case WPAS_MODE_INFRA:
+				if (bss) {
+					ret = os_snprintf(pos, end - pos, "freq=%d MHz\n",
+							  bss->freq);
+				}
+				break;
+			case WPAS_MODE_IBSS:
+			case WPAS_MODE_AP:
+			case WPAS_MODE_P2P_GO:
+				ret = os_snprintf(pos, end - pos, "freq=%d MHz\n",
+						  ssid->frequency);
+				break;
+			default:
+				ret = 0;
+				break;
+			}
+			if (ret < 0 || ret >= end - pos)
+				return pos - buf;
+			pos += ret;
 		}
 
 #ifdef CONFIG_AP
-- 
1.7.4.1

